### ISSUE TITLE
improve diagnostic for `typing.cast` to TypeVar in parameter default

### ISSUE THREAD

**Issue by JasonGrace2282** [astral-sh/ty#2321]
### Summary

The following [playground](https://play.ty.dev/d1ed0fb4-3fb8-4c82-ba65-bac5cc7e5be7) gives a diagnostic of
```
Default value of type `TypeVar` is not assignable to annotated parameter type `D@foo`
```
for the following code:
```py
from typing import TypeVar, cast
D = TypeVar('D')

def foo(x: D = cast(D, None)) -> D:
    return x
```

I'm not quite sure if this should be allowed, or if this usage of `cast` should give off an error, but it would be nice if there was either a correct inference or a better error message!

**Comment by question429** [comment]
To clarify, this doesn’t seem like an issue with typing.cast itself.
A default value for a TypeVar-typed parameter must be valid for all possible bindings of that TypeVar, and cast(D, None) doesn’t change that.
The behavior seems reasonable, but the diagnostic could do a better job explaining why this is rejected.
